library/options.inc.php

```diff
@@ -3032,13 +3236,42 @@ function accumActionConditions($field_id, &$condition_str, &$condarr)
             "id:"       . js_escape($condition['id'])       . ", " .
             "itemid:"   . js_escape($condition['itemid'])   . ", " .
             "operator:" . js_escape($condition['operator']) . ", " .
-            "value:"    . js_escape($condition['value'])    . ", " .
-            "andor:"    . js_escape($andor)                 . "}";
+            "value:"    . js_escape($condition['value'])    . ", ";
+        if ($frow['data_type'] == 15 && strpos($frow['edit_options'], '2') !== false) {
+            // For billing codes handle requirement to display its description.
+            $tmp = explode('=', $action, 2);
+            if (!empty($tmp[1])) {
+                $condition_str .= "valdesc:'" . addslashes(getCodeDescription($tmp[1])) . "', ";
+            }
+        }
+        $condition_str .= "andor:'" . js_escape($andor) . "'}";
```

Comment:
this is likely broken since there are outer quotes. rec removing the outer quotes:
```
$condition_str .= "andor:" . js_escape($andor) . "}";
```